test/unit/strings_spec.lua

```diff
@@ -98,3 +98,40 @@ describe('vim_strnsave_unquoted()', function()
     eq('/Program\\nFiles/sh', vim_strnsave_unquoted('/Program"\\n"Files/sh'))
   end)
 end)
+
+describe("reverse_text", function()
+  local reverse_text = function(str)
+    return helpers.internalize(strings.reverse_text(to_cstr(str)))
+  end
+
+  it("handles empty string", function()
+    eq("", reverse_text(""))
+  end)
+
+  it("handles simple cases", function()
+    eq("a", reverse_text("a"))
+    eq("ba", reverse_text("ab"))
+  end)
+
+  it("handles multibyte characters", function()
+    eq("bα", reverse_text("αb"))
+    eq("Yötön yö", reverse_text("öy nötöY"))
+  end)
+
+  it("handles combining chars", function()
+    local utf8_COMBINING_RING_ABOVE = "\204\138"
+    local utf8_COMBINING_RING_BELOW = "\204\165"
+    eq("bba" .. utf8_COMBINING_RING_ABOVE .. utf8_COMBINING_RING_BELOW .. "aa", 
+       reverse_text("aaa" .. utf8_COMBINING_RING_ABOVE .. utf8_COMBINING_RING_BELOW .. "bb"))
+  end)
+
+  local utf8_REPLACEMENT_CHARACTER = "\239\191\189"
+
+  it("replaces invalid utf with the replacement character U+FFFD", function()
+    eq(utf8_REPLACEMENT_CHARACTER .. "ba", reverse_text("ab\192"))
+  end)
+
+  it("replaces an incomplete utf continuation sequence", function()
+    eq(utf8_REPLACEMENT_CHARACTER .. "ba", reverse_text("ab\194"))
+  end)
+end)
```

Comment:
https://en.wikipedia.org/wiki/UTF-8#Invalid_byte_sequences
I'm wondering what the correct strategy for handling invalid utf is? Not just in `reverse_text` but neovim in general. I made example tests for handling invalid bytes and incomplete continuation seq by replacing them. Another strategy is to error on these. @bfredl ?